pkg/lib/aws/s3.go

```diff
@@ -295,37 +328,98 @@ func (c *Client) UploadMsgpackToS3(obj interface{}, bucket string, key string) e
 	return c.UploadBytesToS3(msgpackBytes, bucket, key)
 }
 
-func (c *Client) ReadMsgpackFromS3(objPtr interface{}, bucket string, key string) error {
-	msgpackBytes, err := c.ReadBytesFromS3(bucket, key)
+func (c *Client) UploadDirToS3(localDirPath string, bucket string, s3Dir string, ignoreFns ...files.IgnoreFn) error {
+	localPaths, err := files.ListDirRecursive(localDirPath, false, ignoreFns...)
```

Comment:
Have you considered setting relative to true:
```
localPaths, err := files.ListDirRecursive(localDirPath, true, ignoreFns...)
```

The prefix trimming can be removed and in the for loop https://github.com/cortexlabs/cortex/blob/d5eed9d8ed3755e8649af2a62d4a5e7ddc16f929/pkg/lib/aws/s3.go#L346

```
key := filepath.Join(s3Dir, localPath)
``` 

because localPath is already relative?